packages/eds-core-react/src/components/Datepicker/DatePicker.tsx

```diff
@@ -133,8 +146,12 @@ export const DatePicker = forwardRef<HTMLDivElement, DatePickerProps>(
         }
       : undefined
 
+    useEffect(() => {
+      if (!defaultValue && !value) setInnerValue(null)
+    }, [defaultValue, value])
+
```

Comment:
Would be nice with a description of this side-effect so that we remember why it is there in the future.